video_worker/__init__.py

```diff
@@ -116,6 +113,7 @@ def run(self):
             logger.error('[VIDEO_WORKER] Invalid Video / VEDA Data')
             return
 
+        print os.path.exists(self.workdir)
         if not os.path.exists(self.workdir):
```

Comment:
@Qubad786 @muzaffaryousaf This was the only way I could get this test to pass, which leads me to believe that there's an issue with the mock_mkdir. Can you provide some insight here?

Reply:
@yro may you please tell which test?

Reply:
edx-video-worker/video_worker/tests/test_video_worker.py:test_run.py
It fails to call 
`if not path_exists:
     self.assertTrue(mock_mkdir.called)`
(L195)

Reply:
@yro this is due to: https://github.com/edx/edx-video-worker/blob/03d9605be95c343ff6e864fde82338a5126abe5d/video_worker/tests/test_video_worker.py#L154

We are expecting 2 `path.exists` calls before this actual (i.e. to be asserted) call, putting print solves it because its also calling `os.path.exists` which corrects the call sequence and things start to work. This test is a mess, let get past from it by taking the whole `mock_exists.side_effect` from `ddt`.

Reply:
@Qubad786 Nice -- thanks! Will do.